### PR TITLE
Check for mismatched temp files. Props to @samrensenhouse

### PR DESCRIPTION
**Pull request by alexmunda**
Fixes `not enough bytes` error

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -130,24 +130,43 @@ const CrawlDependencies = solution => {
...
+            _.difference(allPossibleFiles, solution['main-modules']), {}
          )
       })
 }
 
+// Adapted from checkFiles by https://github.com/samrensenhouse
+const cleanMismatchedTempFiles = (elm_stuff_path, file_type_to_check, companion_file_type) => {
+   const files = GlobFs.readdirSync(`${elm_stuff_path}/**/*.${file_type_to_check}`)
+
+   _.forEach(files, file => {
```
Comment (smerchek) [inline index.js]: could this be promisified to do the "exists" and potential "unlink" in parallel for all files?

Comment (alexmunda) [reply index.js]: I could `promisify` the who;e file but that would require using `util.promisify` to stay sane. Which in turn would require users to have node 8+. [elm-webpack-loader](https://github.com/elm-community/elm-webpack-loader) supports 6+ if we want to maintain parity with them. 😢

Comment (smerchek) [reply index.js]: Ah, that's a good consideration. Even though it wouldn't be as clean, is there value in still using the callback versions of the `fs` package so they can be parallelized?

I'm not sure whether the performance difference is significant or not.

Comment (alexmunda) [reply index.js]: Once we get a testable version published  we can see if the sync access makes it super slow and if so we can try asyncing

**Pull request merged**